modules/fluid_properties/src/fluidproperties/TwoPhaseNCGPartialPressureFluidProperties.C

```diff
@@ -0,0 +1,59 @@
+//* This file is part of the MOOSE framework
+//* https://www.mooseframework.org
+//*
+//* All rights reserved, see COPYRIGHT for full restrictions
+//* https://github.com/idaholab/moose/blob/master/COPYRIGHT
+//*
+//* Licensed under LGPL 2.1, please see LICENSE for details
+//* https://www.gnu.org/licenses/lgpl-2.1.html
+
+#include "TwoPhaseNCGPartialPressureFluidProperties.h"
+#include "IdealRealGasMixtureFluidProperties.h"
+
+registerMooseObject("FluidPropertiesApp", TwoPhaseNCGPartialPressureFluidProperties);
+
+InputParameters
+TwoPhaseNCGPartialPressureFluidProperties::validParams()
+{
+  InputParameters params = TwoPhaseNCGFluidProperties::validParams();
+
+  params.addClassDescription(
+      "Two-phase fluid with single NCG using partial pressure mixture model");
+
+  params.addRequiredParam<UserObjectName>("fp_ncg", "NCG fluid properties object");
+
+  return params;
+}
+
+TwoPhaseNCGPartialPressureFluidProperties::TwoPhaseNCGPartialPressureFluidProperties(
+    const InputParameters & parameters)
+  : TwoPhaseNCGFluidProperties(parameters),
+
+    _fp_ncg(getUserObject<SinglePhaseFluidProperties>("fp_ncg"))
+{
+  _fp_2phase = &_fe_problem.getUserObject<TwoPhaseFluidProperties>(_2phase_name);
+  _fp_vapor_primary = &_fe_problem.getUserObject<SinglePhaseFluidProperties>(getVaporName());
```

Comment:
why not move these up in the initializer's list?

Reply:
then you could const them

Reply:
There's some dumb object creation ordering issue I ran into. I'll investigate.

Reply:
Ok here's what it is. The base class decided that these are `const T *` pointers because there is at least one child that creates these things in the constructor, e.g., `IAPWS95TwoPhaseNCGIdealRealFluidProperties`. Note that I want to deprecate that child, so later your suggestion should be able to be done.